contracts/finality/src/state/public_randomness.rs

```diff
@@ -7,8 +7,8 @@ use cosmwasm_std::Order::{Ascending, Descending};
 use cosmwasm_std::{Deps, StdResult, Storage};
 use cw_storage_plus::{Bound, Map};
 
-/// Map of public randomness values by fp public key and block height
-const PUB_RAND_VALUES: Map<(&[u8], u64), Vec<u8>> = Map::new("pub_rand_values");
+/// Map of public randomness values by block height and fp public key
+const PUB_RAND_VALUES: Map<(u64, &[u8]), Vec<u8>> = Map::new("pub_rand_values");
```

Comment:
I suggest we keep the DB schema as is. Pub rand is a bit different from finality signature because the `block_height` is not consecutive -- FP commits pub rand every `X` rollup blocks, and `X` could be very big. Each FP has different `X`

Reply:
This is much better for pruning, as a range based on height can be used, for efficiency.

The fact that the different each FP can have a different height, or interval between blocks, doesn't have any impact here AFAIK.